js/src/main/scala-2.10/scalajson.ast/JValue.scala

```diff
@@ -97,15 +102,9 @@ object JNumber {
   */
 // Due to a restriction in Scala 2.10, we cant override/replace the default apply method
 // generated by the compiler even when the constructor itself is marked private
-final class JNumber private[ast] (val value: String) extends JValue {
-
-  /**
-    * Javascript specification for numbers specify a [[scala.Double]], so this is the default export method to `Javascript`
-    *
-    * @param value
-    */
-  def this(value: Double) = this(value.toString)
-
+final class JNumber private[ast] (val value: String)(
+    private[ast] val constructedFlag: Int)
```

Comment:
Here's https://github.com/mdedetrich/scalajson/pull/41 that splits `JNumber` up.